src/components/ui/weather-card-control/use-weather-card-control.ts

```diff
@@ -1,28 +1,43 @@
 import { useContext, useState } from 'react';
+
+import axios from 'axios';
+import { api } from '../../../const';
 import { CardsContext } from '../../complex/weather/cards-context';
 
 const useWeatherCardControl = () => {
-  const [loadingCity, setLoadingCity] = useState(false);
   const [isActive, setIsActive] = useState(false);
-  const [isLoading, setLoading] = useState(true);
-  const { setCities, cities } = useContext(CardsContext);
+  const [loadingCity, setLoadingCity] = useState(true);
+  const { cities, setCities } = useContext(CardsContext);
 
   const handleAddSearch = () => {
     setIsActive(true);
   };
 
   const handleOnSubmit = ({ values }: any) => {
-    setCities([...cities, values.search]);
-    setIsActive(false);
+    const addWeather = async () => {
```

Comment:
remove the redundant wrap function, leave only try/catch

Reply:
fixed